### DISCUSSION TITLE
How to get Dehomogenization output from VABs

### DISCUSSION THREAD

**Discussion by sange000000** [wenbinyugroup/.github#24] [VABS Related Discussions] [answered]
![image](https://github.com/user-attachments/assets/309252f6-ed45-4b26-9474-f4939540a1be)
I got this above lines in xml file (correct me if this is wrong, I got this somewhere from the internet)

with this I ran 'prevabs -i .xml -h -v' then 'vabs .sg'
The dehomogenization outputs are not generated. Is there something I'm missing here?

**Comment by unitasium** [comment]
The correct command is

`prevabs -i <filename>.xml -d`

then

`vabs <filename>.sg 1`

**Reply by sange000000** [reply]
Do I need the homogenized output, for executing the run to extract the demogenized ones?

when I run directly 
prevabs -i <filename>.xml -d
then
vabs <filename>.sg 1

I get an error 'cannot open the file <filename>.sg for reading!'
But when I do the above 2 commands after doing these 'prevabs -i <filename>.xml -h ' then 'vabs <filename>.sg', I get the .sg.ELE file generated.
Also could you tell me what is the command to get .SN,.U,.SMN,.S,.EM,.EMN,.EN

**Reply by sange000000** [reply]
[ex_airfoil_r_trial.zip](https://github.com/user-attachments/files/17520188/ex_airfoil_r_trial.zip)
Attaching the files here

**Reply by unitasium** [reply, marked as answer]
Yes. You need to run homogenization first:
```
prevabs -i <filename>.xml -h
vabs <filename>.sg
```
then run recovery:
```
prevabs -i <filename>.xml -d
vabs <filename>.sg 1
```
After this, you should get the local stress and strain files.

**Reply by sange000000** [reply]
Thanks

**Reply by sange000000** [reply]
I still dont get the files : .sg.SN,.U,.SMN,.S,.EM,.EMN,.EN. only the following is generated .sg.ELE, opt,v,v1S,v22
